src/librustc/ty/context.rs

```diff
@@ -841,6 +841,12 @@ impl<'tcx> CommonTypes<'tcx> {
             trait_object_dummy_self: mk(Infer(ty::FreshTy(0))),
         }
     }
+
+    #[inline]
+    pub fn err(&self) -> Ty<'tcx> {
```

Comment:
If you want to use `delay_span_bug`, here's a cool trick: add `#[track_caller]` to this function, then use `std::panic::Location::caller()` to get a location to print for the `delay_span_bug`.

cc @anp